src/app/pages/deposits/deposits.component.ts

```diff
@@ -0,0 +1,76 @@
+import { Deposit } from 'src/app/models/Deposit';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ToastrService } from 'ngx-toastr';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { Subscription } from 'rxjs';
+import { Router } from '@angular/router';
+
+import { DepositsService } from 'src/app/services/deposits/deposits.service';
+import { removeSubscription } from 'src/app/helpers/unsubscribe';
+
+@Component({
+  selector: 'app-deposits',
+  templateUrl: './deposits.component.html',
+  styleUrls: ['./deposits.component.scss']
+})
+export class DepositsComponent implements OnInit, OnDestroy {
+  subscribe: Subscription[] = [];
+  headElements = ['Reference', 'Date', 'Amount'];
+  transactions: Deposit[] = [];
+  startDateVal: string = '';
+  endDateVal: string = '';
+  newResults: any[];
+
+  constructor(
+    private depositsService: DepositsService,
+    private spinner: NgxSpinnerService,
+    private toastrService: ToastrService,
+    private router: Router
+  ) {}
+
+  onFilter() {
```

Comment:
I'm not able to review the functionality at the moment since there's an issue on Heroku

![image](https://user-images.githubusercontent.com/12128153/63673238-86d7a700-c7ec-11e9-8f96-7597d3b9559d.png)


However you do have some `ts-lint` issues on this function,

for instance the 

`date`, `fromDate` and `toDate` are not reassigned, hence you should use `const`

Reply:
You could use the [ts-lint-plugin](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-tslint-plugin) from vscode to catch the linting issues

Reply:
Thanks @Tevinthuku , I have worked on this!